packages/flutter_tools/lib/src/project.dart

```diff
@@ -478,23 +481,33 @@ class AndroidProject extends FlutterProjectPlatform {
     return parent.directory.childDirectory('build');
   }
 
-  Future<void> ensureReadyForPlatformSpecificTooling() async {
+  Future<void> ensureReadyForPlatformSpecificTooling({bool exitOnDeprecation = true}) async {
     if (getEmbeddingVersion() == AndroidEmbeddingVersion.v1) {
       globals.printStatus(
 """
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 Warning
 ──────────────────────────────────────────────────────────────────────────────
 Your Flutter application is created using an older version of the Android
-embedding. It's being deprecated in favor of Android embedding v2. Follow the
+embedding. It is being deprecated in favor of Android embedding v2. Follow the
 steps at
 
 https://flutter.dev/go/android-project-migration
 
-to migrate your project.
+to migrate your project. You may also pass the --ignore-deprecation flag to
+ignore this check and continue with the deprecated v1 embedding. However,
+the v1 Android embedding will be removed in future versions of Flutter.
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 """
       );
+      BuildEvent('deprecated-v1-android-embedding-detected', type: 'deprecation', flutterUsage: globals.flutterUsage).send();
```

Comment:
I'd suggest using one of the pre-existing `type`s instead of creating a new one.